src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -384,7 +384,7 @@ private static OrderedItemDataCollection.Builder ComputeItems(LazyItemList lazyI
                                 break;
                             }
 
-                            string fullPath = FileUtilities.GetFullPath(frag.TextFragment, frag.ProjectDirectory);
+                            string fullPath = FileUtilities.NormalizePathForComparisonNoThrow(frag.TextFragment, frag.ProjectDirectory);
```

Comment:
Is it possible problem that we now add normalized path, but on line 404 we remove based on original value?

I'm completely lost on what exactly is this trying to do, how often and with how much data can it be called - hence cannot access if there might be any concerns with flipping the line 404 to normalize everything as well before attempting to remove. If there might be perf concerns - than maybe we need double indexed lookup - by raw and normalized path?

But maybe I'm completely confused :-)

Reply:
I flipped to this method, because the missed lookup for the update was done by it:
https://github.com/dotnet/msbuild/blob/3ca39400746f9e2234aa45a3baf7a768e30c3d25/src/Build/Evaluation/LazyItemEvaluator.cs#L443
 
To me it looks like a simple bug.

For the question - not sure how Removal works even without my changes, since TextFragment contains a file name only: 
![{03424A8C-1C88-47ED-8E7E-94E47D092BA1}](https://github.com/user-attachments/assets/d48e61d2-a471-40e2-acbd-d3475e6cb94e)

@rainersigwald , do you know any story behind this?